tests/drivers/can/api/src/main.c

```diff
@@ -788,6 +789,7 @@ void test_send_receive_msgq(void)
 	for (i = 0; i < nframes; i++) {
 		err = k_msgq_get(&can_msgq, &frame, TEST_RECEIVE_TIMEOUT);
 		zassert_equal(err, 0, "receive timeout");
+		assert_frame_equal(&frame, &test_std_frame_1, 0);
```

Comment:
This test uses the same test frame with the same number of frames twice, maybe switching to frameid 2 for the second stage would at least make sure the recvq isn't just an unlimited/big number of test_std_frame_1?

Reply:
True, but the API doesn't guarantee an ordering of received frames even in loopback mode. Some controllers will order them FIFO style others may order them by CAN-ID priority. The k_msgq will contain the frames in the order delivered by the controller. The `test_send_receive_msgq()` test case validates that the expected number of frames can be queued.